src/ts/packedrtree.ts

```diff
@@ -84,13 +151,40 @@ export async function* streamSearch(
             const high32Offset = uint32Array[(nodePos << 1) + 9]
             const offset = readUint52(high32Offset, low32Offset);
 
-            if (isLeafNode)
+            if (isLeafNode) {
+                Logger.debug("yielding feature");
                 yield [offset, pos - leafNodesOffset]
-            else
-                queue.push([offset, level - 1])
+                continue;
+            }
+
+            // request up to this many extra bytes if it means we can eliminate
+            // an extra request
+            const combineRequestThreshold = 256 * 1024 / NODE_ITEM_LEN;
+
+            const tail = queue[queue.length - 1];
+            if (tail !== undefined 
+                && tail.level() == next.level() - 1
+                && offset < tail.endNode() + combineRequestThreshold) {
+
+                Logger.debug(`Extending existing node: ${tail}, newOffset: ${tail.endNode()} -> ${offset}`);
+                tail.extendEndNodeToNewOffset(offset);
+                continue;
+            } 
+
+            let newNodeRange: NodeRange = (()=> {
+                let level = next.level() - 1;
+                let range: [number, number] = [offset, offset + 1];
+                return new NodeRange(range, level);
+            })();
+
+            if (tail !== undefined && tail.level() == next.level() - 1) {
+                Logger.debug(`pushing new node at offset: ${offset} rather than merging with distant ${tail}`);
+            } else {
+                Logger.debug(`pushing new level for ${newNodeRange} onto queue with tail: ${tail}`);
+            }
+
+            queue.push(newNodeRange);
         }
-        // order queue to traverse sequential
-        queue.sort((a, b) => b[0] - a[0])
```

Comment:
Since we now `shift` the next node off, rather than `pop` it, our `queue` is ordered by construction, and this `sort` call became unnecessary.

Reply:
Nice!